qiskit/transpiler/passes/layout/sabre_layout.py

```diff
@@ -210,7 +216,18 @@ def run(self, dag):
             self.property_set["layout"] = initial_layout
             self.routing_pass.fake_run = False
             return dag
+
         dist_matrix = self.coupling_map.distance_matrix
```

Comment:
We might want to put this in an else condition because this will probably compute the distance matrix (it is cached so if it was used before it won't) which can be expensive depending on the size of the graph. We probably don't want to do build two distance matrices if we're only going to use one.